### PR TITLE
update README and logging for ilab taxonomy diff

### PR DESCRIPTION
**Pull request by khaledsulayman**
With #2087, we now support running diff against an empty taxonomy. Some of the README and logging was still a bit ambiguous around it so here are some proposed updates.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -591,7 +600,7 @@ Before following these instructions, ensure the existing model you are adding sk
    ilab data generate --endpoint-url http://localhost:8000/v1
    ```
 
-Note that it is also possible to generate a synthetic dataset based on the entire contents of the taxonomy repo using the `--taxonomy-base=empty` option:
+Note that it is also possible to generate a synthetic dataset based on the entire contents of the taxonomy repo using the --taxonomy-base=empty option:
 
    ```shell
    ilab data generate --taxonomy-base=`empty`
```
Comment (jaideepr97) [inline README.md:606]: @khaledsulayman I think we want the quotes removed on L606 not L603

File: README.md [hunk 2]
```diff
@@ -591,7 +600,7 @@ Before following these instructions, ensure the existing model you are adding sk
    ilab data generate --endpoint-url http://localhost:8000/v1
    ```
 
-Note that it is also possible to generate a synthetic dataset based on the entire contents of the taxonomy repo using the `--taxonomy-base=empty` option:
+Note that it is also possible to generate a synthetic dataset based on the entire contents of the taxonomy repo using the --taxonomy-base=empty option:
```
Comment (danmcp) [inline README.md]: Why remove the `` here?

**Review by danmcp — APPROVED** [review]

**Review by jaideepr97 — APPROVED** [review]

**Review by alinaryan — APPROVED** [review]

**Comment by jaideepr97** [comment]
@Mergify.io backport to release-v0.18

**Comment by jaideepr97** [comment]
@Mergifyio backport release-v0.18

**Comment by khaledsulayman** [comment]
https://github.com/Mergifyio backport release-v0.18

**Pull request merged**